core/sequencer/sequencer.go

```diff
@@ -95,50 +95,51 @@ func (s *Sequencer) AddDirectory(dirIDs ...string) {
 	}
 }
 
-// RunBatchForAllMasterships runs RunBatch on all directires this sequencer is currently master for.
-func (s *Sequencer) RunBatchForAllMasterships(ctx context.Context, batchSize int32) error {
-	glog.Infof("RunBatchForAllMasterships")
+// ForAllMasterships runs f once for all directories this server is master for.
+func (s *Sequencer) ForAllMasterships(ctx context.Context, f func(ctx context.Context, dirID string) error) error {
 	cctx, cancel := context.WithCancel(ctx)
 	defer cancel()
 	masterships, err := s.tracker.Masterships(cctx)
 	if err != nil {
 		return err
 	}
-
 	var lastErr error
-	for dirID, whileMaster := range masterships {
+	for dirID, mastershipCtx := range masterships {
+		if err := f(mastershipCtx, dirID); err != nil {
+			lastErr = err
+		}
+	}
+	return lastErr
+}
+
+// RunBatchForAllMasterships runs RunBatch on all directires this sequencer is currently master for.
```

Comment:
```suggestion
// RunBatchForAllMasterships runs RunBatch on all directories this sequencer is currently master for.
```

Reply:
nit: Since `RunBatch` is a method on `KeyTransparencySequencerClient`, rather than `Sequencer`, I'm not sure it's obvious which method you're referring to in this comment. Perhaps it'd be helpful to readers if you either:
- stated that it runs `KeyTransparencySequencerClient.RunBatch`
- described what the result of running `RunBatch` will be, rather than just saying it'll run it, e.g.
```suggestion
// RunBatchForAllMasterships integrates a batch of mutations into each directory for which this sequencer is master.
```
Having said this, your current comment is consistent with the comments on other method in this file.